lib/features/login/data/local/oidc_configuration_cache_manager.dart

```diff
@@ -1,34 +1,54 @@
 import 'package:core/utils/app_logger.dart';
 import 'package:model/oidc/oidc_configuration.dart';
 import 'package:shared_preferences/shared_preferences.dart';
+import 'package:tmail_ui_user/features/caching/clients/oidc_configuration_cache_client.dart';
+import 'package:tmail_ui_user/features/caching/utils/caching_constants.dart';
+import 'package:tmail_ui_user/features/login/data/extensions/oidc_configutation_cache_extension.dart';
 import 'package:tmail_ui_user/features/login/data/network/config/oidc_constant.dart';
 import 'package:tmail_ui_user/features/login/data/network/oidc_error.dart';
+import 'package:tmail_ui_user/features/login/domain/extensions/oidc_configuration_extensions.dart';
 import 'package:tmail_ui_user/main/utils/app_config.dart';
 
 class OidcConfigurationCacheManager {
   final SharedPreferences _sharedPreferences;
+  final OidcConfigurationCacheClient _oidcConfigurationCacheClient;
 
-  OidcConfigurationCacheManager(this._sharedPreferences);
+  OidcConfigurationCacheManager(this._sharedPreferences, this._oidcConfigurationCacheClient);
 
   Future<OIDCConfiguration> getOidcConfiguration() async {
-    final authority = _sharedPreferences.getString(OIDCConstant.keyAuthorityOidc);
-    if (authority == null || authority.isEmpty) {
-      throw CanNotFoundOIDCAuthority();
-    } else {
+    final oidcConfigurationCache = await _oidcConfigurationCacheClient.getItem(CachingConstants.oidcConfigurationCacheKeyName);
+    if (oidcConfigurationCache == null) {
+      final authority = _sharedPreferences.getString(OIDCConstant.keyAuthorityOidc);
```

Comment:
In the previous version, we used `SharedPreferences` to save `keyAuthorityOidc`. But in this part, I have switched to using Hive to save the OIDCConfiguration object as well, so that the old user is not affected, we have to retrieve its value if it exists in `SharedPreferences`.